digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/ManoeuvreService.scala

```diff
@@ -211,28 +218,130 @@ class ManoeuvreService(roadLinkService: RoadLinkService) {
     }
   }
 
-  def createManoeuvre(userName: String, manoeuvre: NewManoeuvre) = {
+  def createManoeuvre(userName: String, manoeuvre: NewManoeuvre, roadlinks: Seq[RoadLink]) : Long = {
     withDynTransaction {
-      dao.createManoeuvre(userName, manoeuvre)
+      createWithoutTransaction(userName, manoeuvre, roadlinks)
     }
   }
 
+  def createWithoutTransaction(userName: String, manoeuvre: NewManoeuvre, roadlinks: Seq[RoadLink]) : Long = {
+    if(!isValid(manoeuvre, roadlinks))
+      throw new InvalidParameterException("Invalid 'manoeuvre")
+
+    dao.createManoeuvre(userName, manoeuvre)
+  }
+
   def deleteManoeuvre(s: String, id: Long): Long = {
     withDynTransaction {
       dao.deleteManoeuvre(s, id)
     }
   }
 
-  def getSourceRoadLinkIdById(id: Long) = {
+  def deleteManoeuvreFromSign(id: Long): Long = {
+    logger.info("expiring manoeuvre")
+    withDynTransaction {
+      dao.deleteManoeuvreByTrafficSign(id)
+    }
+  }
+
+  def getSourceRoadLinkIdById(id: Long) : Long = {
     withDynTransaction {
       dao.getSourceRoadLinkIdById(id)
     }
   }
 
-  def find(id: Long) = {
+  def find(id: Long) : Option[Manoeuvre] = {
     withDynTransaction {
       dao.find(id)
     }
   }
 
+  private def createManoeuvreFromTrafficSign(manouvreProvider: ManoeuvreProvider, newTransaction: Boolean = true): Option[Long] = {
+    logger.info("creating manoeuvre from traffic sign")
```

Comment:
`newTransaction: Boolean = true` could be change by default as  `newTransaction: Boolean = false`